http_service/pulse/consumers.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Library with Pulse factory that can generate Pulse consumers.
+
+Import the ConsumerFactory and generate the connection & consumer you need.
+
+```python
+    # Righ now the factory only has hg_pushes to generate a connection & consumer
+    connection, consumer = ConsumerFactory.hg_pushes(user, password, _on_message)
+    with connection as conn:  # noqa
+        consumer.run()
+```
+"""
+from kombu import Connection, Exchange, Queue
+from kombu.mixins import ConsumerMixin
+
+_CONNECTION_URL = "amqp://{}:{}@pulse.mozilla.org:5671/?ssl=1"
+
+
+def _generate_hg_pushes_queue(user):
+    return Queue(
+        name="queue/{}/pushes".format(user),
+        exchange=Exchange("exchange/hgpushes/v2", type="topic", no_declare=True,),
+        routing_key="#",
+        durable=True,
+        # XXX: This should not be auto delete
```

Comment:
Maybe it makes sense for it to be auto delete? So e.g. if the service is down we don't keep getting messages in the queue

Reply:
If we don't delete the queue then we will keep adding pushes to the Pulse queue and once our Pulse consumer goes back up we will be able to process and catch up.

Reply:
Yep, but we don't want to do that, as those would be old pushes and so probably test selection for them is not needed anymore (and if it is, we'd already have got the request directly via the API).